packages/block-editor/src/components/global-styles/border-panel.js

```diff
@@ -223,6 +276,134 @@ export default function BorderPanel( {
 					/>
 				</ToolsPanelItem>
 			) }
+			{ hasShadowControl && (
+				<>
+					<BaseControl.VisualLabel as="legend">
+						{ __( 'Shadow' ) }
+					</BaseControl.VisualLabel>
+					<ToolsPanelItem
+						label={ __( 'Shadow' ) }
+						hasValue={ hasShadow }
+						onDeselect={ resetShadow }
+						isShownByDefault={ defaultControls.shadow }
+						panelId={ panelId }
+					>
+						<ItemGroup isBordered isSeparated>
+							<ShadowPopover
+								shadow={ shadow }
+								onShadowChange={ setShadow }
+								settings={ settings }
+							/>
+						</ItemGroup>
+					</ToolsPanelItem>
+				</>
+			) }
 		</Wrapper>
 	);
 }
+
+const ShadowPopover = ( { shadow, onShadowChange, settings } ) => {
+	const popoverProps = {
+		placement: 'left-start',
+		offset: 36,
+		shift: true,
+	};
+
+	return (
+		<Dropdown
+			popoverProps={ popoverProps }
+			className="block-editor-global-styles-effects-panel__shadow-dropdown"
+			renderToggle={ renderShadowToggle() }
+			renderContent={ () => (
+				<DropdownContentWrapper paddingSize="medium">
+					<ShadowPopoverContainer
+						shadow={ shadow }
+						onShadowChange={ onShadowChange }
+						settings={ settings }
+					/>
+				</DropdownContentWrapper>
+			) }
+		/>
+	);
+};
+
+function renderShadowToggle() {
+	return ( { onToggle, isOpen } ) => {
+		const toggleProps = {
+			onClick: onToggle,
+			className: classnames( { 'is-open': isOpen } ),
+			'aria-expanded': isOpen,
+		};
+
+		return (
+			<Button { ...toggleProps }>
+				<HStack justify="flex-start">
+					<Icon
+						className="block-editor-global-styles-effects-panel__toggle-icon"
+						icon={ shadowIcon }
+						size={ 24 }
+					/>
+					<FlexItem>{ __( 'Box shadow' ) }</FlexItem>
+				</HStack>
+			</Button>
+		);
+	};
+}
+
+function ShadowPopoverContainer( { shadow, onShadowChange, settings } ) {
+	const defaultShadows = settings?.shadow?.presets?.default;
+	const themeShadows = settings?.shadow?.presets?.theme;
+	const defaultPresetsEnabled = settings?.shadow?.defaultPresets;
+
+	const shadows = [
+		...( defaultPresetsEnabled ? defaultShadows : [] ),
+		...( themeShadows || [] ),
+	];
+
+	return (
+		<div className="block-editor-global-styles-effects-panel__shadow-popover-container">
+			<VStack spacing={ 4 }>
+				<Heading level={ 5 }>{ __( 'Presets' ) }</Heading>
```

Comment:
I think that we should keep "Presets", and also rename "Shadow" to "Box shadow" in the popover toggle:

1. Repetition — Every header down the way there reads "Shadows"; compare:

| Before | After |
| -- | -- |
| <img width="560" alt="image" src="https://github.com/WordPress/gutenberg/assets/1157901/4508060e-0119-4e5e-9a6c-0225deba577c"> | <img width="561" alt="image" src="https://github.com/WordPress/gutenberg/assets/1157901/1ac31efb-111b-44ec-8376-63c5e37bd22f"> |

2. Accuracy — the popover actually displays Shadow presets, and the Shadow control affects Box shadows, not other types of shadows ie. text shadows.

Reply:
cc. @jasmussen

Reply:
@jasmussen can you confirm the right choice here?

| A | B | C |
| --- | --- | --- |
| <img width="580" alt="image" src="https://github.com/WordPress/gutenberg/assets/1935113/9e1547cc-320a-4e47-84bc-43e03c9a3070"> | <img width="563" alt="image" src="https://github.com/WordPress/gutenberg/assets/1935113/1eb97b1d-908a-43a0-adcd-156950232bf0"> | <img width="563" alt="image" src="https://github.com/WordPress/gutenberg/assets/1935113/0d83657f-33ac-4521-bde7-bd041459b9e9"> |

Reply:
Panel title/inspector section title:

* Border & Shadow if both are supported
* Shadow if only that's supported

ItemGroup item title: Drop shadow (because we might support "Inner shadow" in the future as well)

Flyout title should be the same as the ItemGroup title.

Be mindful of the capitalization, we use sentence case ("Drop shadow"), not titlecase ("Drop Shadow").